### PR TITLE
add trm screening

### PR DESCRIPTION
**Pull request by ryangoree**
- Add `useAddressScreening` hook from council
- Added a blank page at the `/void` route.
- Made the app redirect to the `/void` route and show a dialog to tell the user that they're ineligible if the the screen comes back `false`.
  - If they try to hide the dialog in the dev tools and navigate to another page, the router will keep redirecting them back to the `/void` route.

ineligible-wallet

### CODE REVIEW

**Comment by CPSTL** [comment]
Task linked: [CU-2a1xaqy Add TRM screening](https://app.clickup.com/t/2a1xaqy)

**Comment by sentilesdal** [comment]
#### 📝 Changed routes:


| [**/void**](https://elf-frontend-staging-bglfz4hcq-element-finance.vercel.app/void) |
|:-:|
|<a href="https://elf-frontend-staging-bglfz4hcq-element-finance.vercel.app/void">Screenshot of /void</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/wIpKw_LfUKsC8mI1oRk4csvCdrwISz0P6ziwxaF0ho5s2Gxt6vx3hGUsaEBQl8y_bnwUVOYoodd3hOwrtEnSoQ.png">(view full size)</a> |







Commit acbb4c71f492804d15c2258abaddcb2c897385b9 (https://elf-frontend-staging-bglfz4hcq-element-finance.vercel.app).

File: src/ui/wallets/IneligibleAccountDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+}: IneligibleWalletDialogProps): ReactElement {
+  const { darkModeClassName } = useDarkMode();
+  const isSmallScreen = useMedia("(max-width: 639px)");
+  return (
+    <Dialog
+      isCloseButtonShown={false}
+      style={isSmallScreen ? smallScreenStyle : undefined}
+      className={classNames(darkModeClassName, "pb-0", "overflow-auto")}
+      isOpen={isOpen}
+      // onClose={onClose}
```
Comment (DannyDelott) [inline src/ui/wallets/IneligibleAccountDialog.tsx]: dedcode

File: src/ui/wallets/IneligibleAccountDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      title={
+        <span className="text-center text-base py-6">{t`Ineligible Account`}</span>
+      }
+    >
+      <div className="p-5">
+        <Callout intent="danger">
+          This account is not eligible to use this website.
```
Comment (DannyDelott) [inline src/ui/wallets/IneligibleAccountDialog.tsx]: missing ttag

File: src/ui/app/App/App.tsx [hunk 3]
```diff
@@ -33,6 +36,13 @@ const App: FC<AppProps> = ({ children }) => {
     useWeb3React<Web3Provider>();
   const isWrongChain = active && chainId !== AddressesJson.chainId;
   const chainName = ChainNames[AddressesJson.chainId as ChainId];
+  const { route, replace } = useRouter();
+  const { pass } = useAddressScreening(account);
+  const safeReplace = useRef(replace).current;
```
Comment (DannyDelott) [inline src/ui/app/App/App.tsx]: Let's combine this and the useEffect below into a custom hook, eg: `useRedirectToVoidRouteEffect`.

**Review by DannyDelott — APPROVED** [review]
Nice! couple of small nits

**Pull request merged**